### PR TITLE
TST: Add a test for np.pad where constant_values is an object

### PR DESCRIPTION
**Pull request by hmaarrfk**
Relates to https://github.com/numpy/numpy/pull/11358#discussion_r217890065

Padding an array of objects used to be supported.
This PR aims to immortalize that requirement in a test. 

cc: @lagru @eric-wieser

### CODE REVIEW

File: numpy/lib/tests/test_arraypad.py [hunk 1]
```diff
@@ -502,6 +502,16 @@ def test_check_large_integers(self):
         expected = np.full(7, int64_max, dtype=np.int64)
         assert_array_equal(test, expected)
 
+    def test_check_object_array(self):
+        arr = np.empty(1, dtype=object)
+        obj = object()
+        arr = np.pad(arr, pad_width=1, mode='constant', constant_values=obj)
```
[the remark this answers is not in the document]
Comment (hmaarrfk) [reply numpy/lib/tests/test_arraypad.py]: done!

File: numpy/lib/tests/test_arraypad.py [hunk 2]
```diff
@@ -502,6 +502,21 @@ def test_check_large_integers(self):
...
+    def test_check_object_array(self):
+        arr = np.empty(1, dtype=object)
+        obj_a = object()
+        arr[0] = obj_a
+        obj_b = object()
+        obj_c = object()
+        arr = np.pad(arr, pad_width=1, mode='constant',
+                     constant_values=(obj_b, obj_c))
+
+        expected = np.empty((3, ), dtype=object)
```
Comment (eric-wieser) [inline numpy/lib/tests/test_arraypad.py]: A style nit: 1-tuples should be written as `(3,)`, I think

**Review by eric-wieser — APPROVED** [review]
Looks good to me

**Comment by hmaarrfk** [comment]
Thanks Eric, Excited to see this improvement in numpy.

**Comment by mattip** [comment]
Thanks @hmaarrfk

**Pull request merged**